go/src/reader/reader.go

```diff
@@ -1,51 +1,69 @@
 package reader
 
-import "regexp"
-import "fmt"
+import (
+	"fmt"
+	"regexp"
+
+	"types"
+)
 
 func Read_str(input string) string {
-  fmt.Println(tokenizer(input));
-  return "ASdf";
+	fmt.Println(tokenizer(input))
+	return "ASdf"
 }
 
 type TokenReader struct {
-  tokens []string;
-  position int;
+	tokens   []string
+	position int
 }
 
 func (r TokenReader) peek() string {
-  return r.tokens[r.position]
+	return r.tokens[r.position]
 }
 
 func (r TokenReader) next() string {
-  char := r.tokens[r.position]
-  r.position += 1;
-  return char
+	char := r.tokens[r.position]
+	r.position += 1
+	return char
 }
 
 func tokenizer(input string) []string {
-  var token_regex = regexp.MustCompile("[\\s,]*(~@|[\\[\\]{}()'`~^@]|\"(?:\\\\.|[^\\\\\"])*|;.*|[^\\s\\[\\]{}('\"`,;)]*\")");
-  results := make([]string, 0, 1);
+	var token_regex = regexp.MustCompile(`[\s,]*(~@|[\[\]{}()'` + "`" + `~^@]|"(?:\\.|[^\\"])*"|;.*|[^\s\[\]{}('"` + "`" + `,;)]*)`)
+	var results []string
```

Comment:
This is the same as make. I did this originally as I was trying to figure out what was happening, and It is not strictly needed.